main.py

```diff
@@ -150,16 +155,56 @@ def get_active_rules():
         arguments = line.split()
         rule = parse_rule(arguments)
         if rule["name"] and rule["name"] not in dev:
+            rule["ip"] = get_interface_ip(rule["name"])
             rules.append(rule)
             dev.add(rule["name"])
             rules.sort(key=lambda x: x["name"])
     return rules
 
 
+def get_interfaces():
+    try:
+        result = subprocess.run(
+            ["ip", "-o", "-4", "addr", "show"],
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        output = result.stdout
+        interfaces = {}
+        for line in output.split("\n"):
+            if line:
+                parts = line.split()
+                iface = parts[1]
+                ip = parts[3].split("/")[0]
+                interfaces[iface] = ip
+        return interfaces
+    except subprocess.CalledProcessError as e:
+        print(f"Error retrieving interfaces: {e}")
+        return {}
+
+
+def get_interface_ip(interface):
+    try:
+        result = subprocess.run(
+            ["ip", "-o", "-4", "addr", "show"],
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+        output = result.stdout
```

Comment:
These lines are duplicated from above, can we make it a seperate one and reuse in both places

Reply:
added new `run_ip_command` function to handle this